### PR TITLE
Give option to list unselected cases in the Custom Selection component

### PR DESCRIPTION
**Pull request by zhx828**
This solves https://github.com/cBioPortal/cbioportal/issues/6361

This is an improvement pr of https://github.com/cBioPortal/cbioportal-frontend/pull/2589

### CODE REVIEW

File: src/pages/studyView/addChartButton/customCaseSelection/CustomCaseSelection.tsx [hunk 1]
```diff
@@ -76,14 +79,26 @@ export default class CustomCaseSelection extends React.Component<ICustomCaseSele
...
     @autobind
     @action
-    onClick() {
-        let cases = this.props.selectedSamples.map(sample => {
+    onClick(selectMode: SelectMode) {
+        let selectedCases;
+        if (selectMode === SelectMode.SELECTED) {
+            selectedCases = this.props.selectedSamples;
+        } else {
+            const _selectedCaseIds = _.reduce(this.props.selectedSamples, (acc, next) => {
```
Comment (kalletlak) [inline src/pages/studyView/addChartButton/customCaseSelection/CustomCaseSelection.tsx]: may be `_.keyBy(this.props.selectedSamples,sample=>sample.uniqueSampleKey)`

Comment (zhx828) [reply src/pages/studyView/addChartButton/customCaseSelection/CustomCaseSelection.tsx]: @kalletlak I can update. But do they have same performance? Or just for convenience?

Comment (kalletlak) [reply src/pages/studyView/addChartButton/customCaseSelection/CustomCaseSelection.tsx]: just a cleaner way

**Comment by alisman** [comment]
I think the UI is a tiny bit confusing now.    I would suggest something like this it's more clear they are opposites of each other.  

image

**Comment by zhx828** [comment]
@alisman very similar. fontawesome does not have the exact icon in your screenshot
Screen Shot 2019-12-19 at 5 44 18 PM

**Comment by inodb** [comment]
@zhx828 @alisman @kalletlak it seems like this PR didn't go through product review? Let's discuss on engineering meeting how we can avoid these issues

**Pull request merged**